cmake/onnxruntime_providers.cmake

```diff
@@ -304,6 +304,9 @@ if (onnxruntime_USE_TENSORRT OR onnxruntime_USE_DNNL)
   if(APPLE)
     set_property(TARGET onnxruntime_providers_shared APPEND_STRING PROPERTY LINK_FLAGS "-Xlinker -exported_symbols_list ${ONNXRUNTIME_ROOT}/core/providers/shared/exported_symbols.lst")
   elseif(UNIX)
+    if (onnxruntime_USE_TENSORRT)
+      set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -Wno-deprecated-declarations")
```

Comment:
is it safe to override cxx flags globally here? i think previously the hack was done at the top level cmakelists.txt
is the error happening only when building the onnxruntime_providers_tensorrt target?
if so, can we fix it by setting the compile options for that target only?

Reply:
The error happened on Linux only. I move the flag to onnxruntime_providers_tensorrt compile option.